### ISSUE TITLE
Package cannot be installed on M1 Macs

### ISSUE THREAD

**Issue by bryaan** [bicycle1885/CodecZlib.jl#58]
I am trying to install but I get a build error.  Running on M1 Mac.

```
julia> Pkg.build("CodecZlib")
    Building CodecZlib → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/05916673a2627dd91b4969ff8ba6941bc85a960e/build.log`
ERROR: Error building `CodecZlib`:
┌ Warning: Platform `arm64-apple-darwin21.1.0` is not an officially supported platform
└ @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:450
ERROR: LoadError: KeyError: key "unknown" not found
```

**Comment by aryan-jain** [comment]
Facing the same issue. Any resolution?

**Comment by duanasq** [comment]
Facing same issue

**Comment by Ifihan** [comment]
Hi, I'm also facing the same issue

**Comment by hare1039** [comment]
Hi, I have same error on the new mac os

```
ERROR: Error building `CodecZlib`: 
┌ Warning: Platform `arm64-apple-darwin22.4.0` is not an officially supported platform
```

**Comment by ViralBShah** [comment]
Works fine on M1 for me. Please post more details on Julia version and such if still a problem.

**Comment by Ifihan** [comment]
I just tried building and I got the same error as above. Installing it works fine. My Julia version is ` 1.8.1 `

```
Building CodecZlib → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/05916673a2627dd91b4969ff8ba6941bc85a960e/build.log`
ERROR: Error building `CodecZlib`:
┌ Warning: Platform `arm64-apple-darwin21.5.0` is not an officially supported platform
└ @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:450
ERROR: LoadError: KeyError: key "unknown" not found
```

**Comment by ViralBShah** [comment]
What does `Sys.MACHINE` give you? Are you running a macOS beta or something?

cc @staticfloat @giordano

**Comment by ViralBShah** [comment]
BinaryProvider is no longer really used. So somehow you are picking up a very old release of this package, it seems - and it is trying to build some binaries and such.

**Comment by Ifihan** [comment]
`Sys.MACHINE` gives `"arm64-apple-darwin21.5.0"`. No, I am not running a MacOS Beta. I'm currently on Ventura.

**Comment by Ifihan** [comment]
> BinaryProvider is no longer really used. So somehow you are picking up a very old release of this package, it seems - and it is trying to build some binaries and such.

What should I do? Uninstall and re-install the package?

**Comment by ViralBShah** [comment]
Yeah, try to update, or some other package is forcing an old version of this package. We stopped using BinaryProvider years ago. I would recommend at least using the latest Julia 1.8 release (or 1.9 if you can), and updating the packages. You can also look at Pkg.status() to see what might be holding it back.

**Comment by giordano** [comment]
BinaryProvider hasn't been used in this package for over 3 years, since #49. If BinaryProvider is involved here, coming from this package, an extremely old version of the package is being used

**Comment by Ifihan** [comment]
After uninstalling and reinstalling the package, I was able to resolve the issue successfully. In the process, I identified a reason that could be causing this problem.

One of the contributing factors was that some packages were still relying on the `0.6.0` version, which resulted in errors in the M1 architecture. For instance, I had the `StackOverflow` package and it was utilizing the `0.6.0` version of `CodecZlib`. To address this, I experimented with a solution. I first installed the latest version of `CodecZlib` and then added the `StackOverflow` package. Upon inspecting the package statuses, I noticed that the version of `CodecZlib` reverted back to `0.6.0`, accompanied by a warning message

`Info Packages marked with ⌅ have new versions available but cannot be upgraded. To see why use status --outdated -m`

To investigate further, I executed the command mentioned in the warning: `status --outdated`. The outcome of this command was a display similar to the image shown below:

<img width="483" alt="image" src="https://github.com/JuliaIO/CodecZlib.jl/assets/49107589/b8d93d74-3537-4ea1-837e-603763cb7d0b">

As a result, I devised a potential solution to the issue. My approach involves executing the `status --outdated` command to identify which packages are utilizing the outdated version of `CodecZlib` and triggering the error. Subsequently, the `CodecZlib` version in those specific packages to prevent encountering this error in the future.

Thank you @ViralBShah  and @giordano for taking time to attend to this issue.
